Library/Homebrew/cmd/update.rb

```diff
@@ -46,13 +48,47 @@ def update
 
     # automatically tap any migrated formulae's new tap
     report.select_formula(:D).each do |f|
-      next unless (HOMEBREW_CELLAR/f).exist?
+      next unless (dir = HOMEBREW_CELLAR/f).exist?
       migration = TAP_MIGRATIONS[f]
       next unless migration
       tap_user, tap_repo = migration.split "/"
       install_tap tap_user, tap_repo
+      # update tap for each Tab
+      tabs = dir.subdirs.each.map { |d| Tab.for_keg(Keg.new(d)) }
+      tabs.each { |tab| tab.source["tap"] = "#{tap_user}/homebrew-#{tap_repo}" }
+      tabs.each(&:write)
```

Comment:
I may find another bug here. You should only update tab if current installation is from a core formula.